WordPress/gutenberg
# Framework: Drop deprecations slated for 3.1 removal

This pull request seeks to remove deprecations slated for removal in the upcoming 3.1.0 release.

Specifically, this includes:

 - All components in `wp.blocks.*` are removed. Please use `wp.editor.*` instead.
 - `wp.blocks.withEditorSettings` is removed. Please use the data module to access the editor settings `wp.data.select( "core/editor" ).getEditorSettings()`.
 - All DOM utils in `wp.utils.*` are removed. Please use `wp.dom.*` instead.
 - `isPrivate: true` has been removed from the Block API. Please use `supports.inserter: false` instead.
 - `wp.utils.isExtraSmall` function removed. Please use `wp.viewport.isExtraSmall` instead.
 - `getEditedPostExcerpt` selector removed (`core/editor`). Use `getEditedPostAttribute( 'excerpt' )` instead.

See: https://wordpress.org/gutenberg/handbook/reference/deprecated/#3-1-0

__Testing instructions:__

Verify that there are no regressions in impacted behavior, and importantly that no references to deprecated behaviors exist in core code.

Notably, test the excerpt field, which had not been previously upgraded until this pull request.
